addon/components/link-to-component.js

```diff
@@ -15,6 +16,11 @@ export default LinkComponent.extend({
 
     let owner = getOwner(this);
 
+    assert(
+      `You attempted to use a link-to within a routeless engine, this is not supported.`,
```

Comment:
So it turns out that the above lookup didn't work as  `config:environment` isn't listed in the owners registry. I'm sure it will be possible to retrieve the `modulePrefix`, though I'm unsure how exactly.

Either way, I've updated the assertions message to include a note about the `{{link-to-external}}` component and a link to its documentation.